### PR TITLE
client certificate auth in netty server

### PR DESCRIPTION
**Pull request by andrey-vasilyev**

### CODE REVIEW

File: ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/EngineConnectorConfig.kt [hunk 1]
```diff
@@ -126,8 +128,9 @@ class EngineSSLConnectorBuilder(
     override var keyStore: KeyStore,
     override var keyAlias: String,
     override var keyStorePassword: () -> CharArray,
-    override val privateKeyPassword: () -> CharArray,
-    override val requiredClientCACertificates: List<X509Certificate> = emptyList()
+    override var privateKeyPassword: () -> CharArray,
 ) : EngineConnectorBuilder(ConnectorType.HTTPS), EngineSSLConnectorConfig {
     override var keyStorePath: File? = null
+    override var trustStore: KeyStore? = null
+    override var trustStorePath: File? = null
```
Comment (hfhbd) [inline ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/EngineConnectorConfig.kt:135]: Personally, I don't understand a useful use case with trustStore and trustStorePath. I don't know, why keyStorePath is available...

Comment (andrey-vasilyev) [reply ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/EngineConnectorConfig.kt:135]: I guess you can figure this out from keyStorePath usage or ask ktor team. Personally I believe it is there for convenience but of course there might be stronger incentives.

**Comment by hfhbd** [comment]
@andrey-vasilyev Sorry for the delay, I moved.

**Review by hfhbd — APPROVED** [review]

File: ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/EngineConnectorConfig.kt [hunk 2]
```diff
@@ -126,8 +128,9 @@ class EngineSSLConnectorBuilder(
     override var keyStore: KeyStore,
     override var keyAlias: String,
     override var keyStorePassword: () -> CharArray,
-    override val privateKeyPassword: () -> CharArray,
-    override val requiredClientCACertificates: List<X509Certificate> = emptyList()
+    override var privateKeyPassword: () -> CharArray,
```
Comment (hfhbd) [inline ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/EngineConnectorConfig.kt:131]: I thing, this should be fixed in a seperate KTOR-Issue.

**Review by hfhbd — APPROVED** [review]

**Review by hfhbd — APPROVED** [review]

File: ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/EngineConnectorConfig.kt (hunk 2 above)
Comment (hfhbd) [reply ktor-server/ktor-server-host-common/jvm/src/io/ktor/server/engine/EngineConnectorConfig.kt]: Created https://youtrack.jetbrains.com/issue/KTOR-1085

**Pull request merged**